services/web/server/src/simcore_service_webserver/projects/projects_handlers.py

```diff
@@ -521,3 +521,112 @@ async def get_active_project(request: web.Request) -> web.Response:
 
     except ProjectNotFoundError as exc:
         raise web.HTTPNotFound(reason="Project not found") from exc
+
+
+@login_required
+@permission_required("project.node.create")
+async def create_node(request: web.Request) -> web.Response:
+    user_id = request[RQT_USERID_KEY]
+    project_uuid = request.match_info.get("project_id")
+    body = await request.json()
+
+    try:
+        # ensure the project exists
+
+        await projects_api.get_project_for_user(
+            request.app,
+            project_uuid=project_uuid,
+            user_id=user_id,
+            include_templates=True,
+        )
+        data = {
+            "node_id": await projects_api.add_project_node(
```

Comment:
is that not already in projects_node_handlers.py ???
is that why you needed to change the codeclimate exception list? did you remove it from projects_node_handlers.py? why not keeping these separated?

Reply:
You are correct. I have no idea how this stuff ended up in the `projects_handerls.py`.
Code was moved refactored.